lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py

```diff
@@ -159,9 +163,13 @@ def update_fields(p):
         v = params.pop(old_k)
         params_update[new_k] = v
 
-    extra_vars = params.get('extra_vars_path')
+    extra_vars = str(params.get('extra_vars'))
```

Comment:
Judging from the test failure, you probably need to do the string coercion inside of the block `if extra_vars is not None`